addons/point_of_sale/static/src/js/models.js

```diff
@@ -746,7 +748,11 @@ exports.PosModel = Backbone.Model.extend({
         var self = this;
         this.set('synch',{ state: 'connecting', pending: orders.length});
 
-        return self._save_to_server(orders, options).done(function (server_ids) {
+        // Need to deep copy orders to avoid deleting new orders (which are saved to "orders" in localStorage also)
+        // while previous orders are in progress of "_save_to_order" and removed them after "_save_to_order" successfull
+        var flush_orders = JSON.parse(JSON.stringify(orders));
```

Comment:
@jorenvo: thanks for your comments above, i total agree all. I will try to improve this too :)